bubble_sort.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+def bubble_sort(arr)
+  arr.each {
+    x = 0
+    while x < arr.length - 1
+      if arr[x] > arr[x + 1]
+        tmp = arr[x + 1]
+        arr[x + 1] = arr[x]
+        arr[x] = tmp
+      end
+      x += 1
+    end
+  }
+end
+
+def bubble_sort_by(arr)
+  arr.each {
+    x = 0
+    while x < arr.length - 1
+      if arr[x].length > arr[x + 1].length
+        tmp = arr[x + 1]
+        arr[x + 1] = arr[x]
+        arr[x] = tmp
+      end
+      x += 1
+    end
+  }
+end
```

Comment:
* [ ] bubble_sort_by raises an error if block is not given. 
- **[HINT]** Check for `block_given?` at the beginning of the method and raise an error if no block was passed.